### PR TITLE
Use from_parent

### LINKED ISSUE #7: Fix PytestDeprecationWarning
Thank you for this invaluable package. Lately I see this message:

```
venv/lib/python3.8/site-packages/pytest_mccabe.py:44: PytestDeprecationWarning: direct construction of McCabeItem has been deprecated, please use McCabeItem.from_parent
    return McCabeItem(path, parent, complexity)
```

Please see this [**similar issue with comments**](https://github.com/carsongee/pytest-pylint/issues/120) providing additional info.

### PR DESCRIPTION
**Pull request by MartinThoma**
Closes #7

### CODE REVIEW

**Comment by MartinThoma** [comment]
This is the first time I touch a pytest plugin, so please review it before merging :pray:

**Comment by asmeurer** [comment]
It seems this plugin is very similar to pytest flakes. Do you know which one came first? That plugin has the same issue https://github.com/fschulze/pytest-flakes/issues/26.

I tried applying your fix here to pytest-flakes, replacing `complexity` with` flakesignore`, but I get the error 

```
TypeError: from_parent() got an unexpected keyword argument 'flakesignore'
```

But I remove `flakesignore` from the super.from_parent call, I get

```
TypeError: __init__() missing 1 required positional argument: 'flakesignore'
```

File: pytest_mccabe.py [hunk 1]
```diff
@@ -41,7 +41,13 @@ def pytest_collect_file(path, parent):
     if config.option.mccabe and path.ext == '.py':
         complexity = config._mccabe_complexities(path)
         if complexity != 0:
-            return McCabeItem(path, parent, complexity)
+            if hasattr(McCabeItem, "from_parent"):
```
Comment (The-Compiler) [inline pytest_mccabe.py]: I don't think there's a good reason to still support pytest < 5.4.0. IMHO it'd be fine to drop it and only support newer versions.

File: pytest_mccabe.py [hunk 2]
```diff
@@ -57,15 +63,22 @@ class McCabeError(Exception):
...
+        super(McCabeItem, self).__init__(fspath, parent)
         self._mtime = None
         if hasattr(self, 'add_marker'):
             self.add_marker("mccabe")
         else:
             self.keywords["mccabe"] = True
         self.complexity = complexity
 
+    @classmethod
+    def from_parent(cls, parent, fspath, **kwargs):
```
Comment (The-Compiler) [inline pytest_mccabe.py]: Could you explain why this override is needed? I think it should be possible to pass additional arguments into `from_parent` and get them in `__init__` without anything like that, no?

**Review by The-Compiler — CHANGES_REQUESTED** [review]
Thanks! I added some comments for possible simplifications.

**Comment by The-Compiler** [comment]
> It seems this plugin is very similar to pytest flakes. Do you know which one came first?

pytest-flakes is from 2013, pytest-mccabe from 2015. And indeed it's based on pytest-flakes' code!

> I tried applying your fix here to pytest-flakes, replacing `complexity` with` flakesignore`

Could you show your full patch somewhere (e.g. in a draft PR for pytest-flakes)?

**Comment by asmeurer** [comment]
> Could you show your full patch somewhere (e.g. in a draft PR for pytest-flakes)?

@fschulze is transferring ownership of pytest-flakes to me. Once that happens, I will push up what I have (but it's basically this PR except with the `complexity` flag replaced with `flakesignore`).

**Comment by asmeurer** [comment]
See the most recent commit at https://github.com/asmeurer/pytest-flakes/pull/30

File: pytest_mccabe.py (hunk 1 above)
Comment (MartinThoma) [reply pytest_mccabe.py]: Cool - I removed that part :+1:

File: pytest_mccabe.py (hunk 2 above)
Comment (MartinThoma) [reply pytest_mccabe.py]: I don't remember why I thought it was necessary. I think the renaming of path to fspath was, but not the override and I struggled a bit to find out ... I removed the override :-)

**Comment by MartinThoma** [comment]
I see travis failing, but when I execute pytest locally it runs just fine. Do you have any idea why?

To the issue itself: As I don't understand the "from_parent" (which is the core part of this PR), I don't understand the issue.

**Comment by The-Compiler** [comment]
Thanks for the update! I think it's failing because only old pytest versions get tested, which are now not supported anymore.

I'll need to update the CI configuration (and probably also switch to GitHub Actions while I'm at it) so that this actually gets tested against a newer pytest version. It'll likely take me a couple of days (or perhaps weeks) though, as I'm busy with some other stuff at the moment. If you have time to help with that as well, let me know and I can give you some pointers, though!

**Comment by MartinThoma** [comment]
Ah, I see. I'm not super excited about fixing CI pipelines ... let's see, maybe I'll give it a try :smile: 

Should I make another PR for that or would it fit best in here?

**Comment by The-Compiler** [comment]
I realized I could almost copy-paste a config I wrote for [pytest-xvfb](https://github.com/The-Compiler/pytest-xvfb/), so that's what I ended up doing in b5f2cd6767034e3565e242a6dbddd2b33691a9aa. I also cleaned up various code and updated supported versions in 42c7fc1f998b4198ae1e69dabfdcc633a82fc833.

Thanks for the fix and for getting the ball rolling again! So from my point of view, things would be ready for a 2.0 release - any objections?

**Comment by MartinThoma** [comment]
Oh, nice! Thanks for merging :-)

**Comment by The-Compiler** [comment]
You're welcome! So, anything missing from your point of view or anything else you'd want to contribute before a release? Otherwise I'd push a v2.0 tag and hope the automation takes care of the rest :)

**Comment by MartinThoma** [comment]
It looks fine to me - but as I said, take this with caution. That is the first pytest plugin I'm contributing to :-)

**Comment by The-Compiler** [comment]
Alright! I published [pytest-mccabe 2.0](https://pypi.org/project/pytest-mccabe/) - if anything is amiss, a v2.1 is just a version bump, commit, tag and "git push" away as the [automation](https://github.com/The-Compiler/pytest-mccabe/runs/900108309?check_suite_focus=true) worked out just fine :tada:

**Pull request merged**